### PR TITLE
EditPayment Component

### PR DESCRIPTION
**Pull request by rishkwal**
Component Details:

- [x] AddPayment component routes directly to the EditPayment component
- [x] Route: `/clients/:clientId/payments/:paymentId/update`
- [x] Allows to change details of any payment
- [x] Allows user to generate a Bitcoin invoice for the payment
- [x] Checks and displays the status of the Bitcoin invoice
- [x] Displays the checkout link for Bitcoin invoice
- [x] Cancel button sends back to `/clients/:clientId`
- [x] Add loading when `Update Payment` is clicked
- [x] Only users with `BTC` or `sats` as their currency are shown the option to view Invoice

### CODE REVIEW

**Comment by rishkwal** [comment]
For reference: https://www.loom.com/share/7f47f68eec1942318b6b7270e49f6ab6

File: src/pages/EditPaymentPage.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import Typography from '@material-ui/core/Typography'
+
+import EditPayment from '../components/EditPaymentForm'
+
+class EditPaymentPage extends React.Component {
+
+    render() {
+
+        const { clientId } = this.props.match.params
+        const { paymentId } = this.props.match.params
```
Comment (otech47) [inline src/pages/EditPaymentPage.js:13]: NIT: combined destructure like
```
{ clientId, paymentId } = this.props.match.params
```

File: src/components/EditPaymentForm.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                <Grid item>
+                    <Button 
+                        variant='contained'
+                        color='inherit'
+                        onClick={cancelEditPayment}
+                    >
+                        {`Done`}
+                    </Button>
+                </Grid>
+                <BtcInvoiceModal open={openInvoice} onClose={handleCloseInvoice} bitcoinCheckoutUrl={bitcoinCheckoutUrl}/>
```
Comment (otech47) [inline src/components/EditPaymentForm.js:282]: NIT: indent props here
```
<BtcInvoiceModal
    open={openInvoice}
    onClose={handleCloseInvoice}
    bitcoinCheckoutUrl={bitcoinCheckoutUrl}
/>
```

File: src/components/BtcInvoiceModal.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    return (
+        <>
+            <Modal
+                open={props.open}
+                onClose={props.onClose}
+            >
+                <Box sx={modalStyle}>
+                    <iframe title='invoice' height='660px' src={`${props.bitcoinCheckoutUrl}`}>
+                    </iframe>
```
Comment (otech47) [inline src/components/BtcInvoiceModal.js:19]: NIT iframe I believe don't need a closing tag

File: src/components/AddPaymentForm.js [hunk 4]
```diff
@@ -71,11 +71,12 @@ const AddPaymentForm = (props) => {
             }
             const newPayment = await createPayment({ variables })
             if (loadingNewPayment) return <LoadingProgress/>
-            if (newPayment.errors) {
+            if (errorNewPayment) {
```
Comment (otech47) [inline src/components/AddPaymentForm.js:74]: Let's check if we can move this down to the catch clause so our error handling is cleaner/consistent

**Review by otech47 — CHANGES_REQUESTED** [review]
let's address the error handling cleanup and then merge

**Pull request merged**